tests/PHPUnit/System/ArchiveInvalidationTest.php

```diff
@@ -128,6 +140,70 @@ public function testAnotherApi()
         }
     }
 
+    public function testDisablePluginArchive()
+    {
+        $config = Config::getInstance();
+        $config->General['disable_archiving_segment_for_plugins'] = 'testPlugin';
+        $this->assertTrue(Rules::isSegmentPluginArchivingDisabled('testPlugin'));
+
+        $config->General['disable_archiving_segment_for_plugins'] = ['testPlugin', 'testPlugin2'];
+        $this->assertTrue(Rules::isSegmentPluginArchivingDisabled('testPlugin'));
+
+        $config->General['disable_archiving_segment_for_plugins'] = 'testPlugin,testPlugin2';
+        $this->assertTrue(Rules::isSegmentPluginArchivingDisabled('testPlugin2'));
+
+        $config->General['disable_archiving_segment_for_plugins'] = '';
+        $this->assertFalse(Rules::isSegmentPluginArchivingDisabled('testPlugin'));
+
+    }
+
+    public function testDisablePluginArchiveCaseInsensitive()
+    {
+        $config = Config::getInstance();
+        $config->General['disable_archiving_segment_for_plugins'] = 'testplugin,testplugin2';
+        $this->assertTrue(Rules::isSegmentPluginArchivingDisabled('testPlugin'));
+    }
+
+    public function testDisablePluginArchiveSpecialCharacters()
+    {
+        //special characters will not work
+        $config = Config::getInstance();
+        $config->General['disable_archiving_segment_for_plugins'] = '!@##$%^^&&**(()_+';
+        $this->assertFalse(Rules::isSegmentPluginArchivingDisabled('!@##$%^^&&**(()_+'));
+    }
+
+    public function testDisablePluginArchiveBySiteId()
+    {
+        //test siteId 1 by string
+        Config::setSetting('General_1', 'disable_archiving_segment_for_plugins', 'testPlugin');
+        $this->assertTrue(Rules::isSegmentPluginArchivingDisabled('testPlugin',1));
+
+        //test siteId 1 by array
+        Config::setSetting('General_1', 'disable_archiving_segment_for_plugins',['testPlugin', 'testPlugin2'] );
+        $this->assertTrue(Rules::isSegmentPluginArchivingDisabled('testPlugin',1));
+
+        //test siteId 1 by string with comma
+        Config::setSetting('General_1', 'disable_archiving_segment_for_plugins','testPlugin,testPlugin2' );
+        $this->assertTrue(Rules::isSegmentPluginArchivingDisabled('testPlugin',1));
+
+        //test empty
+        Config::setSetting('General_1', 'disable_archiving_segment_for_plugins','' );
+        $this->assertFalse(Rules::isSegmentPluginArchivingDisabled('testPlugin',1));
+
+        //test siteId 2 not affect siteId1
+        Config::setSetting('General_2', 'disable_archiving_segment_for_plugins','testPlugin' );
+        $this->assertFalse(Rules::isSegmentPluginArchivingDisabled('testPlugin',1));
+
+
+        //test general setting not affect siteId1
+        Config::setSetting('General', 'disable_archiving_segment_for_plugins','myPlugin' );
+        Config::setSetting('General_1', 'disable_archiving_segment_for_plugins','testPlugin' );
+        $this->assertFalse(Rules::isSegmentPluginArchivingDisabled('myPlugin',1));
+
```

Comment:
Maybe a test to ensure no partial matches work would be good:

```suggestion

        Config::setSetting('General_1', 'disable_archiving_segment_for_plugins', 'testPlugin2');
        $this->assertFalse(Rules::isSegmentPluginArchivingDisabled('testPlugin', 1));
```